File: src/main/resources/Authentication.js
function println(log) {
    var System = Java.type('java.lang.System');
    System.out.println(log);
}

function Authentication() {

    this.handle = function (event) {
        println('Javascript Object: ' + this);
        println('Javascript Event: ' + event);
        println('Javascript username: ' + event.body.username);
        var response = {logged: true, 'groups': ['admin', 'user']};
        return response;
    };

    // http://localhost:9876/api/sayHi/?name=john&surname=wick
    this.sayHi = function (request) {
        var Objects = Java.type('java.util.Objects');
        println('Javascript Object: ' + Objects.hashCode(this));
        println('Javascript request: ' + request);
        return {"say": "Hi! " + request["params"]["name"] + " " + request["params"]["surname"]};
    }

}

function instance() {
    return new Authentication();
}

File: src/main/resources/Authentication.java
import com.fererlab.event.Event;
import com.fererlab.event.Request;
import com.fererlab.service.Service;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class Authentication implements Service {

    @Override
    public Object handle(Event event) {
        System.out.println("Java under resources object: " + this);
        System.out.println("Java under resources Event: " + event);
        System.out.println("under resources username: " + event.getBody().get("username"));
        Map<String, Object> response = new HashMap<>();
        response.put("under resources", true);
        response.put("logged", true);
        response.put("groups", Arrays.asList("admin", "user"));
        return response;
    }

    public Object sayHi(Request request) {
        System.out.println("Java object: " + this);
        System.out.println("Java Request: " + request);
        Map<String, Object> response = new HashMap<>();
        response.put("say", "Hi " + request.getParams().get("name"));
        return response;
    }

}
